### ISSUE TITLE
setPreviewData not triggering runtime call of getStaticProps as per Next.js documentation

### ISSUE THREAD

**Issue by afenton90** [serverless-nextjs/serverless-next.js#1361]
### Describe the bug
If you request a page which has `getStaticProps` with the preview mode cookies set (via `res.setPreviewData`), then `getStaticProps` will be called at request time (instead of at build time). As per https://nextjs.org/docs/advanced-features/preview-mode#step-2-update-getstaticprops.

This is not happening when Next.js is deployed via `@sls-next/cdk-construct` to AWS.

### Actual behaviour
`res.setPreviewData` is called and cookies are set successfully in the browser, however on the page where `getStaticProps` should be called at runtime to trigger a preview, the stale data is served instead.

The page with `getStaticProps` defined also has `revalidate` prop defined at a value of `60`.

### Expected behaviour
Preview mode should be enabled for pages with `getStaticProps` defined as per the Next.js documentation.

### Steps to reproduce
1. Create Next.js app with preview mode and single route with `getStaticProps`.
2. Deploy application using `@sls-next/cdk-construct@3.2.0-alpha.12`.
3. Enable preview mode and go to route which has  `getStaticProps` and preview mode enabled.

### Versions

- OS/Environment: AWS
- @sls-next/cdk-construct version: 3.2.0-alpha.12
- Next.js version: 11.0.1

### Checklist

- [x] You have reviewed the README and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [x] Please first try using the most recent `latest` or `alpha` `@sls-next/serverless-component` [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue or implemented the feature you are trying to use. Note that the old `serverless-next.js` component and the `serverless-next.js` plugin are deprecated and no longer maintained.

**Comment by jvarho** [comment]
How did you verify that `getStaticProps` was not getting called? Did you add logging to verify that the stale data isn't due to something else? The end-to-end tests do test that `getStaticProps` gets called in preview mode, so it has to be a more specific circumstance where it doesn't.

**Comment by afenton90** [comment]
After doing some more tests I found this preview issue only happens when the route has `getStaticProps` defined and `revalidate` as mentioned in the original description. Without `revalidate` defined preview mode works. However, it is my understanding that preview mode should work regardless of whether `revalidate` is defined on the route.

I couldn't find any E2E tests trying this specific scenario?

**Comment by jvarho** [comment]
No e2e tests for that, but I did try locally deploying a create-next-app project with:

```
export const getStaticProps = async (context) => {
  console.log('getStaticProps')
  return {
    props: {},
    revalidate: 60
  }
}
```

The console statement shows up in logs when using preview mode, regardless of revalidate. So something odd is going on. Can you share a full test case?

**Comment by b3nThomas** [comment]
We're having a similar issue, however it's only affects pages where `getStaticProps` _and_ `getStaticPaths` are being used together.
- `@sls-next/serverless-component 3.2.0-alpha.29`
- `next ^10.2.3`

**Comment by afenton90** [comment]
Yeah @b3nThomas that is the same situation for me. @jvarho does that provide enough context to explore this issue further?

**Comment by patricktyndall** [comment]
Hi all -- I created a [reproduction](https://github.com/serverless-nextjs/serverless-next.js/issues/1527#issuecomment-898694278) for #1527 which is a preview mode issue. I'm not sure if it's exactly the same issue, but I think it might be.

Is anyone here using optional catch-alls (`[[`)? If so, could you check if preview mode is actually working for the "base" of your catch all?

**Comment by jvarho** [comment]
This should have been fixed in #1527. Please reopen if it persist with latest alpha.
